src/syft/lib/python/dict.py

```diff
@@ -82,7 +82,11 @@ def id(self) -> UID:
         return self._id
 
     def upcast(self) -> dict:
-        return dict(self)
+        # recursively upcast
+        new_dict = dict()
```

Comment:
NITPICK: Why not?
```
new_dict = {k: upcast(v) for v in self.items()}
```

Reply:
I think the issue I had with list spooked me, so im skeptical we won't hit this issue again with the internal generators... lets see.